utils/regr/regr_syy.cpp

```diff
@@ -129,6 +129,8 @@ mcsv1_UDAF::ReturnCode regr_syy::evaluate(mcsv1Context* context, static_any::any
     if (N > 0)
     {
         long double var_popy = (data->sumy2 - (data->sumy * data->sumy / N));
+        if (var_popy < 0)  // might be -0
```

Comment:
Could I ask you @dhall-MariaDB to make the non-positive checks consistent? Previously you use LEQ and here strict L.

Reply:
Notice that when '<= 0' is used, it is to take an action (almost always return). This is the action defined for '== 0' and we're using '<= 0' to catch micro negative numbers. In these cases, any negative value should be 0 but rounding error in the intermediate may fuzz it. When we use '< 0', it is to convert all micro negative values to 0 and then we continue processing.. This is a different application so consistency exists separately for these two uses.